pyiceberg/io/pyarrow.py

```diff
@@ -2102,3 +2100,100 @@ def _dataframe_to_data_files(
                 for batches in bin_pack_arrow_table(partition.arrow_table_partition, target_file_size)
             ]),
         )
+
+
+@dataclass(frozen=True)
+class TablePartition:
```

Comment:
```suggestion
class _TablePartition:
```
One minor request: I prefer to make this private because this class primarily serves as an intermediate state of our internal logic of write support

Reply:
Noted.